test/Npgsql.Tests/CommandTests.cs

```diff
@@ -348,6 +348,366 @@ public void CloseConnectionWithException()
+                    Assert.That(conn.ExecuteScalar("SELECT COUNT(*) FROM pg_prepared_statements"), Is.EqualTo(0));
+
+                    using (var cmd = new NpgsqlCommand("INSERT INTO data (name) VALUES (:p0);", conn))
+                    {
+                        cmd.IsPersistent = true; 
+
+                        cmd.Parameters.Add(new NpgsqlParameter("p0", NpgsqlDbType.Text));
+                        cmd.Prepare();
+                        cmd.Parameters["p0"].Value = "test1";
+                        using (var dr = cmd.ExecuteReader())
+                        {
+                            Assert.IsNotNull(dr);
+                            dr.Close();
+                            Assert.That(dr.RecordsAffected, Is.EqualTo(1));
+                        }
+                        Assert.That(conn.ExecuteScalar("SELECT COUNT(*) FROM pg_prepared_statements"), Is.EqualTo(1));
+                    }
+                    Assert.That(conn.ExecuteScalar("SELECT COUNT(*) FROM data WHERE name = 'test1'"), Is.EqualTo(1));
+                    Assert.That(conn.ExecuteScalar("SELECT COUNT(*) FROM pg_prepared_statements"), Is.EqualTo(1), "Persistent prepared statement deallocated");
+
+                    string stmtName = conn.ExecuteScalar("SELECT name FROM pg_prepared_statements LIMIT 1") as string;
+
+                    // Rerun the test using the persistent prepared statement
+                    using (var cmd = new NpgsqlCommand("INSERT INTO data (name) VALUES (:p0);", conn))
+                    {
+                        cmd.IsPersistent = true;
+
+                        cmd.Parameters.Add(new NpgsqlParameter("p0", NpgsqlDbType.Text));
+                        cmd.Prepare();
+                        cmd.Parameters["p0"].Value = "test2";
+                        using (var dr = cmd.ExecuteReader())
+                        {
+                            Assert.IsNotNull(dr);
+                            dr.Close();
+                            Assert.That(dr.RecordsAffected, Is.EqualTo(1));
+                        }
+                        Assert.That(conn.ExecuteScalar("SELECT COUNT(*) FROM pg_prepared_statements"), Is.EqualTo(1));
+                    }
+                    Assert.That(conn.ExecuteScalar("SELECT COUNT(*) FROM data WHERE name = 'test2'"), Is.EqualTo(1));
+                    Assert.That(conn.ExecuteScalar("SELECT COUNT(*) FROM pg_prepared_statements"), Is.EqualTo(1), "Persistent prepared statement deallocated");
+
+                    Assert.That(conn.ExecuteScalar("SELECT name FROM pg_prepared_statements LIMIT 1") as string, Is.EqualTo(stmtName), "Persistent prepared statement name changed unexpectedly");
+                }
+            }
+            finally
+            {
+                // Clear the pools so that we dont mess up other tests that depend
+                // on the clean state of pg_prepared_statements
+                NpgsqlConnection.ClearAllPools();
```

Comment:
This could interfere with tests running in parallel. It's better to have an `Application Name` in the connection string and to only clear the pool for that connection string.